From df1eee23e082309900db7caf2acbc36c0190df59 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Fri, 15 Oct 2021 07:56:51 +0000
Subject: [PATCH] GCC: fix ambiguous reference to AudioSourceCallback

GCC sees two definitions:
1. class media::AudioOutputStream::AudioSourceCallback
   media::AudioOutputStream::AudioSourceCallback::AudioSourceCallback
2. class media::AudioOutputStream::AudioSourceCallback

Use explicit reference to avoid confusion.
---
 services/audio/device_listener_output_stream.h  | 3 ++-
 services/audio/device_listener_output_stream.cc | 3 ++-
 1 file changed, 4 insertions(+), 2 deletion(-)

diff --git a/services/audio/device_listener_output_stream.h b/services/audio/device_listener_output_stream.h
index 3dccbfd..8ebd737 100644
--- a/services/audio/device_listener_output_stream.h
+++ b/services/audio/device_listener_output_stream.h
@@ -34,7 +34,8 @@ class DeviceListenerOutputStream final
 
   // AudioOutputStream implementation
   bool Open() final;
-  void Start(AudioSourceCallback* source_callback) final;
+  void Start(media::AudioOutputStream::AudioSourceCallback*
+                 source_callback) final;
   void Stop() final;
   void SetVolume(double volume) final;
   void GetVolume(double* volume) final;
diff --git a/services/audio/device_listener_output_stream.cc b/services/audio/device_listener_output_stream.cc
index 7e5985d..5e52978 100644
--- a/services/audio/device_listener_output_stream.cc
+++ b/services/audio/device_listener_output_stream.cc
@@ -34,7 +34,8 @@ bool DeviceListenerOutputStream::Open() {
   return stream_->Open();
 }
 
-void DeviceListenerOutputStream::Start(AudioSourceCallback* source_callback) {
+void DeviceListenerOutputStream::Start(
+    media::AudioOutputStream::AudioSourceCallback* source_callback) {
   DCHECK(task_runner_->BelongsToCurrentThread());
   DCHECK(!source_callback_);
   DCHECK(source_callback);
-- 
2.32.0
